python/ray/includes/serialization.pxi

```diff
@@ -513,3 +515,36 @@ cdef class RawSerializedObject(SerializedObject):
                              MEMCOPY_THREADS)
         else:
             memcpy(&buffer[0], self.value_ptr, self._total_bytes)
+
+cdef class ArrowSerializedObject(SerializedObject):
+    cdef:
+        object value
+        int64_t _total_bytes
+
+    def __init__(self, value):
+        super(ArrowSerializedObject,
+              self).__init__(ray_constants.OBJECT_METADATA_TYPE_ARROW)
+        self.value = value
+        sink = pa.MockOutputStream()
+        writer = pa.ipc.new_stream(sink, self.value.schema)
+        writer.write(self.value)
```

Comment:
Seems unfortunate we basically write it twice.

Reply:
This is used to get the size, so that we can allocate memory according to it, and avoid writing to bytes then copy to the buffer. MockOutputStream does not actually write anything. It only calculates the size.